zipline/data/bundles/alpha_vantage_api.py

```diff
@@ -0,0 +1,323 @@
+    asset_list = list_assets()
+    sid = 0
+
+    for symbol in asset_list:
+      try:
+        df = av_get_data_for_symbol(symbol, start, end, interval)
+        
+        first_traded = df.index[0]
+        auto_close_date = df.index[-1] + pd.Timedelta(days=1)
+
+        if 'split' in df.columns:
+            split = calc_split(sid, df)
+            divs_splits['splits'] = divs_splits['splits'].append(split)
+
+        if 'dividend' in df.columns:
+            div = calc_dividend(sid, df)
+            divs_splits['divs'] = pd.concat([divs_splits['divs'], div])
+
+        yield (sid, df), symbol, start, end, first_traded, auto_close_date, exchange
+
+      except KeyboardInterrupt:
+        exit()
+
+      except Exception as e:
+        print(e)
+
+      sid = sid + 1
+
+def metadata_df():
+    metadata_dtype = [
+        ('symbol', 'object'),
+        # ('asset_name', 'object'),
+        ('start_date', 'datetime64[ns]'),
+        ('end_date', 'datetime64[ns]'),
+        ('first_traded', 'datetime64[ns]'),
+        ('auto_close_date', 'datetime64[ns]'),
+        ('exchange', 'object'), ]
+    metadata_df = pd.DataFrame(
+        np.empty(len(list_assets()), dtype=metadata_dtype))
+
+    return metadata_df
+
+
+@bundles.register('alpha_vantage', calendar_name="NYSE", minutes_per_day=390)
+def api_to_bundle(interval=['1m']):
+    def ingest(environ,
+               asset_db_writer,
+               minute_bar_writer,
+               daily_bar_writer,
+               adjustment_writer,
+               calendar,
+               start_session,
+               end_session,
+               cache,
+               show_progress,
+               output_dir
+               ):
+
+        divs_splits = {'divs': pd.DataFrame(columns=['sid', 'amount',
+                                                  'ex_date', 'record_date',
+                                                  'declared_date', 'pay_date']),
+                       'splits': pd.DataFrame(columns=['sid', 'ratio',
+                                                    'effective_date'])}
+
+        def minute_data_generator():
+            return (sid_df for (sid_df, *metadata.iloc[sid_df[0]]) in 
+                df_generator(interval='1m', start=start_session, end=end_session, divs_splits=divs_splits))
+
+        def daily_data_generator():
+            return (sid_df for (sid_df, *metadata.iloc[sid_df[0]]) 
+                in df_generator(interval='1d', start=start_session, end=end_session, divs_splits=divs_splits))
+
+        metadata = metadata_df()
+
+        for _interval in interval:
+            if _interval == '1d':
+                daily_bar_writer.write(daily_data_generator(), show_progress=True)
+            elif _interval == '1m':
+                minute_bar_writer.write(minute_data_generator(), show_progress=True)
+
+        # convert back wrong datatypes after pd.concat
+        divs_splits['splits']['sid'] = divs_splits['splits']['sid'].apply(lambda x: np.int(x))
```

Comment:
generally `apply` is not efficient. you should avoid it whenever possible. this is better:
`divs_splits['splits']['sid'].astype(np.int)`